### PR TITLE
ft: S3C-3194 bucket notification processor

### PR DESCRIPTION
**Pull request by naren-rajendran**
New queue processor for bucket notifications. This is a basic extension of existing processor component.

### CODE REVIEW

File: extensions/notification/queueProcessor/QueueProcessor.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                    { error: err.message });
+                return undefined;
+            }
+            if (options && options.disableConsumer) {
+                this.emit('ready');
+                return undefined;
+            }
+            const groupId =
+                `${this.notifConfig.queueProcessor.groupId}-${this.site}`;
+            const concurrency = this.notifConfig.queueProcessor.concurrency;
```
Comment (jonathan-gramain) [inline extensions/notification/queueProcessor/QueueProcessor.js]: You can use object deconstruction `const { groupId, concurrency } = this.notifConfig.queueProcessor` to shorten the code.

**Review by jonathan-gramain — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e approve

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
@bert-e approve

**Comment by naren-rajendran** [comment]
@bert-e reset

**Pull request merged**